lib/Hakyll/Web/Template/Internal.hs

```diff
@@ -203,12 +204,16 @@ applyTemplate' tes context x = go tes
 
     ----------------------------------------------------------------------------
 
-    applyStringExpr wrap msg expr = mapError wrap (applyExpr expr) >>= getString
+    applyStringExpr :: Maybe String -> String -> TemplateExpr -> Compiler String
+    applyStringExpr mbPrepend msg expr =
+        prepending (applyExpr expr) >>= getString
```

Comment:
I'm not sure any more why I passed `mapError` into `applyStringExpr` - just to prevent code duplication? Or really to not get the error message when `getString` fails, but only when `applyExpr` fails? Could've been a bug, could've been intentional.
So maybe we should just call `prependErrorMessage` at the call site, simplifying `applyStringExpr`.

Reply:
Having another look at the code again, I do remember now: it was intentional. By nesting the `withErrorMessage` calls, one does get told the error location twice in the "stack trace":
```
Hakyll.Web.Template.applyTemplate: expected string but got xy for expr '$…$'
In expr '$…$'
In …
```
Sorry for a) leading you astray with my comment b) not writing test cases for this...